src/sentry/tagstore/snuba/backend.py

```diff
@@ -299,22 +323,26 @@ def get_release_tags(self, project_ids, environment_id, versions):
         # NB we add release as a condition rather than a filter because
         # this method is already dealing with version strings rather than
         # release ids which would need to be translated by the snuba util.
-        conditions = [['release', 'IN', versions]]
+        # XXX: This should also be `sentry:release`
+        key = 'release'
+        conditions = [[key, 'IN', versions]]
         aggregations = [
-            ['count()', '', 'count'],
+            ['count()', '', 'times_seen'],
             ['min', SEEN_COLUMN, 'first_seen'],
             ['max', SEEN_COLUMN, 'last_seen'],
         ]
 
-        result = snuba.query(start, end, ['release'], conditions, filters, aggregations)
+        # XXX: This doesn't group by project ID so isn't a 1:1 translation of
+        # the existing implementation
```

Comment:
I haven't really put any thought into what to do about this yet, but right now I think this will cause problems.

Reply:
I think you can just add 'project_id` to the groupby columns no? 

```
snuba.query(start, end, ['release', 'project_id'], ...)
```

Reply:
I think so for the query itself. I'll also need to add the project ID to the different type objects (probably will do that first for the legacy backend and then rebase this onto that change.)

Reply:
Actually, method (`get_release_tags`) is only used in a single API endpoint: https://github.com/getsentry/sentry/blob/c62a7b251ee9fc840a81621374ba34bf46a7ceb5/src/sentry/api/serializers/models/release.py#L171-L178

The end result is basically what this method is already doing (combining release data by version across all projects.) I do think it's worth making this behavior consistent across all of the different tagstore implementations so they're directly comparable/swappable, so there are two options on how to move forward here:

1. Make each backend return a single object per release, and remove the reduction logic from the API endpoint.
    * This backend (Snuba) would keep doing what it is already doing here.
    * The v2 (Citus) version will have to continue doing the reduction in the application like the API endpoint is doing (since the data is partitioned by project, the aggregates can't be pushed down to the database.)
    * The legacy version could do either SQL aggregates or combine results in the application like the API endpoint is doing, whichever ends up being simpler.
2. Keep the existing return type (multiple objects per release.) The only change would be to add the project to the `GROUP BY` (like @alex-hofsteede mentioned above), but this wouldn't actually have to include the project ID in the returned objects like I had previously thought.

Reply:
Ended up going with option 2 because it's simpler and I'm lazy.